From e5bb6c65f143e2c49f9a7b81205f620b8974c343 Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Sun, 21 Nov 2021 20:14:14 +0100
Subject: [PATCH] Allow devendoring of pybind11

CMakeLists.txt:
Add the option `BESPOKE_SYSTEM_PYBIND11` to indicate that a system
installed pybind11 should be used instead of the bundled version.

libs/CMakeLists.txt:
Depending on `BESPOKE_SYSTEM_PYBIND11` use the pkgconfig integration to
find a system installed pybind11 or use the bundled pybind11.
---
 CMakeLists.txt      | 1 +
 libs/CMakeLists.txt | 9 ++++++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf0f0b81b..02e9881ed 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,7 @@ set(BESPOKE_PYTHON_ROOT "" CACHE STRING "Python search path. Override as needed.
 set(BESPOKE_VST2_SDK_LOCATION "" CACHE STRING "Steinberg VST2 SDK directory for non-FOSS builds")
 set(BESPOKE_ASIO_SDK_LOCATION "" CACHE STRING "Steinberg ASIO SDK directory for non-FOSS builds")
 option(BESPOKE_SYSTEM_JSONCPP "Use system-wide installation of jsoncpp" OFF)
+option(BESPOKE_SYSTEM_PYBIND11 "Use a system installation of pybind11" OFF)
 
 # Global CMake options
 set(CMAKE_CXX_EXTENSIONS OFF)
diff --git a/libs/CMakeLists.txt b/libs/CMakeLists.txt
index 5f92ef30d..e85f66bf3 100644
--- a/libs/CMakeLists.txt
+++ b/libs/CMakeLists.txt
@@ -8,7 +8,14 @@ add_subdirectory(psmove EXCLUDE_FROM_ALL)
 add_subdirectory(push2 EXCLUDE_FROM_ALL)
 
 set(PYBIND11_NOPYTHON TRUE)
-add_subdirectory(pybind11 EXCLUDE_FROM_ALL)
+if(NOT BESPOKE_SYSTEM_PYBIND11)
+  message(STATUS "Using bundled pybind11")
+  add_subdirectory(pybind11 EXCLUDE_FROM_ALL)
+else()
+  message(STATUS "Using system installed pybind11")
+  include(FindPkgConfig)
+  find_package(pybind11 COMPONENTS REQUIRED)
+endif()
 add_library(bespoke_pybind11_wrapper INTERFACE)
 target_compile_definitions(bespoke_pybind11_wrapper INTERFACE PYBIND11_EXPORT=)
 target_link_libraries(bespoke_pybind11_wrapper INTERFACE pybind11::pybind11)
